packages/material-ui/src/FormLabel/FormLabel.js

```diff
@@ -69,6 +70,21 @@ const FormLabel = React.forwardRef(function FormLabel(props, ref) {
     states: ['color', 'required', 'focused', 'disabled', 'error', 'filled'],
   });
 
+  const handleClick = (event) => {
+    if (!id) {
+      return;
+    }
+
+    const select = document.querySelector(`[aria-labelledby~="${id}"]`);
+
+    if (!select) {
+      return;
+    }
+
+    event.preventDefault();
```

Comment:
What would this prevent? I think this was needed for mousedown but not click.

Also: Can you can add `TODO` that this would also (incorrectly) fire if the click came at the end of text-selection? It's a minor inconsistency with native label+select but one that we need to be aware of in case there's some actual issue with that behavior.

Reply:
I think that we can handle the selection directly. `getSelection().isCollapsed` seems enough: https://developer.mozilla.org/en-US/docs/Web/API/Selection/isCollapsed.

Reply:
There's a "?" behind IE11 in the compat table for isCollapsed.

Reply:
Tested, IE 11 supports it :)

<img width="258" alt="Capture d’écran 2020-04-29 à 21 09 15" src="https://user-images.githubusercontent.com/3165635/80636416-b87cce00-8a5d-11ea-84f3-ba0b46f41e55.png">